orttraining/orttraining/test/python/orttraining_test_orttrainer_frontend.py

```diff
@@ -524,3 +524,20 @@ def testInstantiateORTTrainer(step_fn):
             assert output_type == _utils.dtype_onnx_to_torch(
                 trainer._onnx_model.graph.output[i].type.tensor_type.elem_type)
 
+    # Save current model as ONNX as a file
+    file_name = os.path.join('..','..','..','temp_onnx_model.onnx')
+    trainer.save_as_onnx(file_name)
+    assert os.path.exists(file_name)
+    with open(file_name, "rb") as f:
+        bin_str = f.read()
+        reload_onnx_model = onnx.load_model_from_string(bin_str)
+    os.remove(file_name)
+
+    # Create a new trainer from persisted ONNX model and compare with original ONNX model
+    trainer_from_onnx = orttrainer.ORTTrainer(reload_onnx_model, model_desc, optim_config)
+    trainer_from_onnx.train_step(data, targets)
+    assert trainer_from_onnx._onnx_model is not None
+    assert (id(trainer_from_onnx._onnx_model) != id(trainer._onnx_model))
+    assert (trainer_from_onnx._onnx_model == trainer._onnx_model)
+    assert (trainer_from_onnx._onnx_model.graph == trainer._onnx_model.graph)
+    assert (onnx.helper.printable_graph(trainer_from_onnx._onnx_model.graph) == onnx.helper.printable_graph(trainer._onnx_model.graph))
```

Comment:
do you want to confirm weighs are indeed updated - if _update_onnx_model_initializers is not called, this test will still pass, right?

Reply:
You are right, we are not comparing weights here, just the structure of the graph. We plan on adding such tests for Checkpointing soon